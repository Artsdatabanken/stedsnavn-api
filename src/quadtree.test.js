const qt = require("./quadtree");

test("quadtile add", async () => {
  const root = {};

  qt.add(root, 0, 0, 0, "A");
  qt.add(root, 0.25, 0.25, 1, "B");
  qt.add(root, 0.25, 0.25, 2, "C");

  expect(root).toMatchSnapshot();
});

test("quadtile find", async () => {
  const root = {};

  qt.add(root, 0, 0, 0, "A");
  qt.add(root, 0.75, 0.75, 1, "B");
  const actual = qt.find2(root, 0.25, 0.75, 0.25, 1);
  expect(actual).toMatchSnapshot();
});

test("quadtile find out of range", async () => {
  const root = {};

  qt.add(root, 0, 0, 0, "A");
  qt.add(root, 0.75, 0.75, 1, "B");
  const actual = qt.find2(root, 0.25, 0.75, 0.24, 1);
  expect(actual).toMatchSnapshot();
});

test("quadtile find root", async () => {
  const root = {};

  qt.add(root, 0, 0, 0, "A");
  qt.add(root, 0.75, 0.75, 1, "B");
  const actual = qt.find2(root, 0.25, 0.75, 0.24, 0);
  expect(actual).toMatchSnapshot();
});

test("quadtile find multi", async () => {
  const root = {};

  qt.add(root, 0.1, 0.1, 2, "A");
  qt.add(root, 0.9, 0.1, 2, "B");
  qt.add(root, 0.1, 0.9, 2, "C");
  qt.add(root, 0.9, 0.9, 2, "D");
  debugger;
  const actual = qt.find2(root, 0.51, 0.75, 0.5, 2);
  expect(actual).toMatchSnapshot();
});
