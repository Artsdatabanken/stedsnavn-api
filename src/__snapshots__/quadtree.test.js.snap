// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quadtile add 1`] = `
Object {
  "nw": Object {
    "se": Object {
      "value": "C",
    },
    "value": "B",
  },
  "value": "A",
}
`;

exports[`quadtile find 1`] = `
Object {
  "dist": 0.8408964152537146,
  "value": "B",
}
`;

exports[`quadtile find multi 1`] = `
Object {
  "dist": 6708203932.49937,
}
`;

exports[`quadtile find out of range 1`] = `
Object {
  "dist": 0.8467453123625271,
  "value": "B",
}
`;

exports[`quadtile find root 1`] = `
Object {
  "dist": 0.8467453123625271,
  "value": "B",
}
`;
